stm32cubemx: wrap with FHSEnv

asmir-abdulahovic opened the pull request:
## Description of changes

As of recently STM32CubeMX is using JxBrowser for, now mandatory, user authentication.
Same browser is extracted from STM32CubeMX install binaries into ~/.stm32cubemx directory and depends on the system shared libraries.

This PR wrapps STM32CubeMX with FHSEnv in order call JxBrowser. 
Package remains mostly the same except a minor change where stdenv is replaced by stdenvNoCC.

STM32CubeMX login page takes a while to appear on Wayland (Sway) but works mostly fine.

Fixes: #272342 

cc @angaz @wucke13 
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

asmir-abdulahovic committed:
stm32cubemx: wrap with FHSEnv

AndersonTorres commented on pkgs/development/embedded/stm32/stm32cubemx/default.nix:
```diff
@@ -1,84 +1,115 @@
-stdenv.mkDerivation rec {
-  pname = "stm32cubemx";
-  version = "6.10.0";
+  pkg = stdenvNoCC.mkDerivation rec {
```
Cryptic abbreviated names are for 80-era C programmers.

Also, it is easy to insert or remove a single `s`.

```suggestion
  package = stdenvNoCC.mkDerivation rec {
```

asmir-abdulahovic replied:
Agree, done.

AndersonTorres commented on pkgs/development/embedded/stm32/stm32cubemx/default.nix:
```diff
@@ -1,84 +1,115 @@
+      sourceProvenance = with sourceTypes; [ binaryBytecode ];
+      license = licenses.unfree;
+      maintainers = with maintainers; [ angaz wucke13 ];
+      platforms = platforms.all;
```
I believe it is a bit misleading, since Java does not run in all platforms.
But I don't think it will be a serious problem.

asmir-abdulahovic replied:
That was a bit out of scope since I focused on bugfix.
Though even more we have reports that the package doesn't work on aarm64 because of upstream issues but I think we should leave platforms revision for later PR if that is fine.

AndersonTorres replied:
OK.

angaz replied:
The app is only released for x86. I think this is a simple thing to fix, and can be done in this PR. It should be good for users to know it won't work on other platforms.

asmir-abdulahovic replied:
Missed this.
OK, I changed it to x86_64-linux only.

angaz reviewed (approved):
Tested running the application using `nixpkgs-review`.
Tested creating a new project, and generating the code. The login window opened, it didn't take in my key presses very well. I had to type the same key multiple times before it accepted a key. Very annoying for the password, but I would consider this a bug in the app, not the nix configuration.

From what I can tell, it's working as expected, and the mentioned bug is resolved.

asmir-abdulahovic commented:
> Tested running the application using `nixpkgs-review`. Tested creating a new project, and generating the code. The login window opened, it didn't take in my key presses very well. I had to type the same key multiple times before it accepted a key. Very annoying for the password, but I would consider this a bug in the app, not the nix configuration.
> 
> From what I can tell, it's working as expected, and the mentioned bug is resolved.

  I found copy/paste to be best solution for credentials input.
  I have a feeling those issues are related to GPU acceleration in JxBrowser i.e. Chromium.
  It's possible to disable hwaccel by using chrome  cmdline flags, but exec string is hardcoded in STM32CubeMX and since login is required only once I think it's not worth to persue further.

Thanks for testing and review.

angaz reviewed (approved):
Cool. Thanks for updating the `platforms`.
Everything looks good to me.

I'm not certain if the commit history is correct. Should changing the `platforms` be in it's own commit?

asmir-abdulahovic commented:
> Cool. Thanks for updating the `platforms`. Everything looks good to me.
> 
> I'm not certain if the commit history is correct. Should changing the `platforms` be in it's own commit?

I squashed it, but I guess it would make more sense to have separate commit now that I think more about it.
Thanks for approwing it again, so I'll not touch it anymore.

angaz commented:
> I squashed it, but I guess it would make more sense to have separate commit now that I think more about it. Thanks for approving it again, so I'll not touch it anymore.

Yeah, my comment was more for future reviewers. Maybe it's something they would want.

asmir-abdulahovic committed:
stm32cubemx: change platforms from all to x86_64-linux

asmir-abdulahovic commented:
> > I squashed it, but I guess it would make more sense to have separate commit now that I think more about it. Thanks for approving it again, so I'll not touch it anymore.
> 
> Yeah, my comment was more for future reviewers. In case they don't see that it's been changed. Maybe it's something they would want.

Understand, added separate commit.

angaz reviewed (approved):
Cool. Thanks
